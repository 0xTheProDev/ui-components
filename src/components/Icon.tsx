import * as React from 'react';
import { IconType } from '../types/icons';

export type IconProps = {
  type: IconType;
};

const Icon: React.SFC<IconProps> = ({ type }) => (
  <i className={`sg-icon sg-icon-${type}`} />
);

export default Icon;
