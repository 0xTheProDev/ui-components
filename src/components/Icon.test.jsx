import React from 'react';
import { shallow } from 'enzyme';
import Icon, { icons } from './Icon';

describe('Icon', () => {
  it('should match the snapshot', () => {
    Object.keys(icons).forEach((icon) => {
      const wrapper = shallow(<Icon type={icon} />);
      expect(wrapper).toMatchSnapshot();
    });
  });
});
