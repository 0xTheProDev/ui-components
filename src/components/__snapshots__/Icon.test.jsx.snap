// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon should match the snapshot 1`] = `
<i
  className="sg-icon sg-icon-airplane is-size-20"
/>
`;

exports[`Icon should match the snapshot 2`] = `
<i
  className="sg-icon sg-icon-airplane-fill is-size-20"
/>
`;

exports[`Icon should match the snapshot 3`] = `
<i
  className="sg-icon sg-icon-align-center is-size-20"
/>
`;

exports[`Icon should match the snapshot 4`] = `
<i
  className="sg-icon sg-icon-align-left is-size-20"
/>
`;

exports[`Icon should match the snapshot 5`] = `
<i
  className="sg-icon sg-icon-align-right is-size-20"
/>
`;

exports[`Icon should match the snapshot 6`] = `
<i
  className="sg-icon sg-icon-analytics is-size-20"
/>
`;

exports[`Icon should match the snapshot 7`] = `
<i
  className="sg-icon sg-icon-builder is-size-20"
/>
`;

exports[`Icon should match the snapshot 8`] = `
<i
  className="sg-icon sg-icon-button is-size-20"
/>
`;

exports[`Icon should match the snapshot 9`] = `
<i
  className="sg-icon sg-icon-calendar is-size-20"
/>
`;

exports[`Icon should match the snapshot 10`] = `
<i
  className="sg-icon sg-icon-caret is-size-20"
/>
`;

exports[`Icon should match the snapshot 11`] = `
<i
  className="sg-icon sg-icon-check is-size-20"
/>
`;

exports[`Icon should match the snapshot 12`] = `
<i
  className="sg-icon sg-icon-check-circle is-size-20"
/>
`;

exports[`Icon should match the snapshot 13`] = `
<i
  className="sg-icon sg-icon-check-circle-fill is-size-20"
/>
`;

exports[`Icon should match the snapshot 14`] = `
<i
  className="sg-icon sg-icon-clean-ui is-size-20"
/>
`;

exports[`Icon should match the snapshot 15`] = `
<i
  className="sg-icon sg-icon-clock is-size-20"
/>
`;

exports[`Icon should match the snapshot 16`] = `
<i
  className="sg-icon sg-icon-code is-size-20"
/>
`;

exports[`Icon should match the snapshot 17`] = `
<i
  className="sg-icon sg-icon-columns is-size-20"
/>
`;

exports[`Icon should match the snapshot 18`] = `
<i
  className="sg-icon sg-icon-contacts is-size-20"
/>
`;

exports[`Icon should match the snapshot 19`] = `
<i
  className="sg-icon sg-icon-contacts-alt is-size-20"
/>
`;

exports[`Icon should match the snapshot 20`] = `
<i
  className="sg-icon sg-icon-copy is-size-20"
/>
`;

exports[`Icon should match the snapshot 21`] = `
<i
  className="sg-icon sg-icon-create is-size-20"
/>
`;

exports[`Icon should match the snapshot 22`] = `
<i
  className="sg-icon sg-icon-csv is-size-20"
/>
`;

exports[`Icon should match the snapshot 23`] = `
<i
  className="sg-icon sg-icon-desktop-view is-size-20"
/>
`;

exports[`Icon should match the snapshot 24`] = `
<i
  className="sg-icon sg-icon-divider is-size-20"
/>
`;

exports[`Icon should match the snapshot 25`] = `
<i
  className="sg-icon sg-icon-drag is-size-20"
/>
`;

exports[`Icon should match the snapshot 26`] = `
<i
  className="sg-icon sg-icon-gear is-size-20"
/>
`;

exports[`Icon should match the snapshot 27`] = `
<i
  className="sg-icon sg-icon-image is-size-20"
/>
`;

exports[`Icon should match the snapshot 28`] = `
<i
  className="sg-icon sg-icon-images is-size-20"
/>
`;

exports[`Icon should match the snapshot 29`] = `
<i
  className="sg-icon sg-icon-image-text is-size-20"
/>
`;

exports[`Icon should match the snapshot 30`] = `
<i
  className="sg-icon sg-icon-info-circle is-size-20"
/>
`;

exports[`Icon should match the snapshot 31`] = `
<i
  className="sg-icon sg-icon-key is-size-20"
/>
`;

exports[`Icon should match the snapshot 32`] = `
<i
  className="sg-icon sg-icon-list is-size-20"
/>
`;

exports[`Icon should match the snapshot 33`] = `
<i
  className="sg-icon sg-icon-locked is-size-20"
/>
`;

exports[`Icon should match the snapshot 34`] = `
<i
  className="sg-icon sg-icon-mail is-size-20"
/>
`;

exports[`Icon should match the snapshot 35`] = `
<i
  className="sg-icon sg-icon-mobile-view is-size-20"
/>
`;

exports[`Icon should match the snapshot 36`] = `
<i
  className="sg-icon sg-icon-pencil is-size-20"
/>
`;

exports[`Icon should match the snapshot 37`] = `
<i
  className="sg-icon sg-icon-people is-size-20"
/>
`;

exports[`Icon should match the snapshot 38`] = `
<i
  className="sg-icon sg-icon-pie-chart is-size-20"
/>
`;

exports[`Icon should match the snapshot 39`] = `
<i
  className="sg-icon sg-icon-plus is-size-20"
/>
`;

exports[`Icon should match the snapshot 40`] = `
<i
  className="sg-icon sg-icon-reload is-size-20"
/>
`;

exports[`Icon should match the snapshot 41`] = `
<i
  className="sg-icon sg-icon-save-draft is-size-20"
/>
`;

exports[`Icon should match the snapshot 42`] = `
<i
  className="sg-icon sg-icon-search is-size-20"
/>
`;

exports[`Icon should match the snapshot 43`] = `
<i
  className="sg-icon sg-icon-segment is-size-20"
/>
`;

exports[`Icon should match the snapshot 44`] = `
<i
  className="sg-icon sg-icon-social is-size-20"
/>
`;

exports[`Icon should match the snapshot 45`] = `
<i
  className="sg-icon sg-icon-sort-asc is-size-20"
/>
`;

exports[`Icon should match the snapshot 46`] = `
<i
  className="sg-icon sg-icon-sort-desc is-size-20"
/>
`;

exports[`Icon should match the snapshot 47`] = `
<i
  className="sg-icon sg-icon-spacer is-size-20"
/>
`;

exports[`Icon should match the snapshot 48`] = `
<i
  className="sg-icon sg-icon-teammate is-size-20"
/>
`;

exports[`Icon should match the snapshot 49`] = `
<i
  className="sg-icon sg-icon-text is-size-20"
/>
`;

exports[`Icon should match the snapshot 50`] = `
<i
  className="sg-icon sg-icon-text-only is-size-20"
/>
`;

exports[`Icon should match the snapshot 51`] = `
<i
  className="sg-icon sg-icon-trash is-size-20"
/>
`;

exports[`Icon should match the snapshot 52`] = `
<i
  className="sg-icon sg-icon-unlocked is-size-20"
/>
`;

exports[`Icon should match the snapshot 53`] = `
<i
  className="sg-icon sg-icon-user-security is-size-20"
/>
`;

exports[`Icon should match the snapshot 54`] = `
<i
  className="sg-icon sg-icon-view is-size-20"
/>
`;

exports[`Icon should match the snapshot 55`] = `
<i
  className="sg-icon sg-icon-warning is-size-20"
/>
`;

exports[`Icon should match the snapshot 56`] = `
<i
  className="sg-icon sg-icon-warning-triangle is-size-20"
/>
`;

exports[`Icon should match the snapshot 57`] = `
<i
  className="sg-icon sg-icon-x is-size-20"
/>
`;

exports[`Icon should match the snapshot 58`] = `
<i
  className="sg-icon sg-icon-x-legacy is-size-20"
/>
`;
